app/helpers/sessions_helper.rb

```diff
@@ -3,18 +3,43 @@ def log_in(user)
     session[:user_id] = user.id
   end
 
+  def remember(user)
+    user.remember
+    cookies.permanent.encrypted[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
   def current_user
-    if session[:user_id]
-      @current_user ||= User.find_by(id: session[:user_id])
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.encrypted[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
     end
   end
 
+
   def logged_in?
     !current_user.nil?
   end
 
   def log_out
+    forget(current_user)
     reset_session
     @current_user = nil
   end
+
+  # Log in as a particular user
+  def log_in_as(user)
```

Comment:
what is this exactly? :D

Reply:
That was a part of the tutorial I did not get to finish yet, will be used later :D